classes/ThankYous/ThankYouFactory.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Claromentis\ThankYou\ThankYous;
+
+use Claromentis\Core\Localization\Lmsg;
+use Claromentis\ThankYou\Exception\ThankYouAuthor;
+use Date;
+use InvalidArgumentException;
+use User;
+
+class ThankYouFactory
+{
+	private $lmsg;
+
+	public function __construct(Lmsg $lmsg)
+	{
+		$this->lmsg = $lmsg;
+	}
+
+	/**
+	 * @param User|int  $author
+	 * @param string    $description
+	 * @param Date|null $date_created
+	 * @return ThankYou
+	 * @throws ThankYouAuthor - If the Author could not be loaded.
+	 */
+	public function Create($author, ?Date $date_created, string $description)
```

Comment:
Very small thing: I'd elect for `$date_created` to come after `$description`. Indeed, it's currently just nullable instead of optional, but in this case it might as well be optional.

The same could be said for the `ThankYou` constructor. I feel it makes more sense to have the date afterwards.

Reply:
Fixed in 46de8caf35065016e43ede1cb605411b0f55f109